integration/operator_context_test.go

```diff
@@ -0,0 +1,212 @@
+package integration
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/TykTechnologies/tyk-operator/api/model"
+	"github.com/TykTechnologies/tyk-operator/api/v1alpha1"
+	"github.com/matryer/is"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestOperatorContextCreate(t *testing.T) {
+	opCreate := features.New("Operator Context").
+		Assess("Create", func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string)
+			is := is.New(t)
+
+			// create api definition
+			_, err := createTestAPIDef(ctx, testNS, envConf)
+			is.NoErr(err) // failed to create apiDefinition
+
+			_, err = createTestOperatorContext(ctx, testNS, envConf)
+			is.NoErr(err) // failed to create operatorcontext
+
+			time.Sleep(reconcileDelay)
+
+			resp, err := http.Get("http://localhost:7000/httpbin/get")
+			is.NoErr(err)
+			is.Equal(resp.StatusCode, 200)
+
+			return ctx
+		}).Feature()
+
+	testenv.Test(t, opCreate)
+}
+
+func TestOperatorContextDelete(t *testing.T) {
+	delApiDef := features.New("Operator Context Delete").
+		Assess("Delete Api Defintion", func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string)
+			is := is.New(t)
+			client := envConf.Client()
+
+			// create operator context
+			operatorCtx, err := createTestOperatorContext(ctx, testNS, envConf)
+			is.NoErr(err) // failed to create operatorcontext
+
+			// create api definition
+			apiDef, err := createTestAPIDef(ctx, testNS, envConf)
+			is.NoErr(err) // failed to create apiDefinition
+
+			err = retryOperation(time.Hour, reconcileDelay, func() error {
+				var opCtx v1alpha1.OperatorContext
+				// shouldn't get deleted
+				err = client.Resources().Get(ctx, operatorCtx.Name, testNS, &opCtx)
```

Comment:
I think these errors should be redeclared/renamed, to avoid confusion and/or shadowing.
```suggestion
			errRetry := retryOperation(time.Hour, reconcileDelay, func() error {
				var opCtx v1alpha1.OperatorContext
				// shouldn't get deleted
				errGet := client.Resources().Get(ctx, operatorCtx.Name, testNS, &opCtx)
```